cmd/check_spec_utils.go

```diff
@@ -16,18 +16,366 @@ package cmd
+	for _, os := range oss {
+		if os == nil {
+			return errOperationStatusNullPointer
+		}
+
+		if isEmpty(os.Status) {
+			return errOperationStatusEmptyStatus
+		}
+	}
+
+	return nil
+}
+
+func validateOperationTypes(ots []string) error {
+	for _, ot := range ots {
+		if isEmpty(ot) {
+			return errOperationTypeEmpty
+		}
+	}
+
+	return nil
+}
+
+func validateErrors(errors []*types.Error) error {
+	for _, err := range errors {
+		if err == nil {
+			return errErrorEmpty
+		}
+
+		if isNegative(int64(err.Code)) {
+			return errErrorNegativeCode
+		}
+
+		if isEmpty(err.Message) {
+			return errErrorEmptyMessage
+		}
+	}
+
+	return nil
+}
+
+func validateCallMethods(methods []string) error {
+	for _, m := range methods {
+		if isEmpty(m) {
+			return errCallMethodEmptyName
+		}
+	}
+
+	return nil
+}
+
+func validateBalanceExemptions(exs []*types.BalanceExemption) error {
+	for _, e := range exs {
+		if e == nil {
+			return errBalanceExemptionEmpty
+		}
+	}
+
+	return nil
+}
+
+func validateBalances(balances []*types.Amount) error {
+	for _, b := range balances {
+		if err := validateBalance(b); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func validateBalance(amt *types.Amount) error {
+	if isEmpty(amt.Value) {
+		return errBalanceEmptyValue
+	}
+
+	if err := validateCurrency(amt.Currency); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func validateCurrency(currency *types.Currency) error {
+	if isEmpty(currency.Symbol) {
+		return errCurrencyEmptySymbol
+	}
+
+	if isNegative(int64(currency.Decimals)) {
+		return errCurrencyNegativeDecimals
+	}
+
+	return nil
+}
+
+func validateCoins(coins []*types.Coin) error {
+	for _, coin := range coins {
+		if err := validateCoinIdentifier(coin.CoinIdentifier); err != nil {
+			return err
+		}
+		if err := validateBalance(coin.Amount); err != nil {
+			return err
+		}
 	}
 
-	return errBlockIdentifier
+	return nil
+}
+
+func validateCoinIdentifier(coinID *types.CoinIdentifier) error {
+	if coinID == nil {
+		return errCoinIdentifierNullPointer
+	}
+
+	if isEmpty(coinID.Identifier) {
+		return errCoinIdentifierEmpty
+	}
+
+	return nil
+}
+
+func twoModes() checkSpecOutput {
+	output := checkSpecOutput{
+		api: modes,
+		validation: map[checkSpecRequirement]checkSpecStatus{
+			diffURLs: checkSpecSuccess,
+		},
+	}
+
+	if isEmpty(Config.OnlineURL) ||
+		isEmpty(Config.Construction.OfflineURL) ||
+		isEqual(Config.OnlineURL, Config.Construction.OfflineURL) {
+		setValidationStatus(output, diffURLs, checkSpecFailure)
+	}
+
+	return output
+}
+
+func markAllValidationsFailed(output checkSpecOutput) {
+	for k := range output.validation {
+		output.validation[k] = checkSpecFailure
+	}
+}
+
+func setValidationStatus(output checkSpecOutput, req checkSpecRequirement, status checkSpecStatus) {
+	output.validation[req] = status
+}
+
+func isEmpty(s string) bool {
+	return s == ""
+}
+
+func isNegative(n int64) bool {
+	return n < 0
+}
+
+func isEqual(s1 string, s2 string) bool {
+	return s1 == s2
+}
```

Comment:
Can we move it to some kind of common helper. This can be re-used outside of check spec

Reply:
Yeah makes sense. I'll create a utils.go in /cmd directory for all CLI processes.